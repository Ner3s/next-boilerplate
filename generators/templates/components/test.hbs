import { {{pascalCase name}} } from '.';

import { screen, render } from '@testing-library/react';

describe('<{{pascalCase name}} />', () => {
  it('should render {{pascalCase name}}', () => {
    render(<{{pascalCase name}} />);

    expect(
      screen.getByRole('heading', {
        name: /{{pascalCase name}}/i
      })
    ).toBeInTheDocument();
  });
});
