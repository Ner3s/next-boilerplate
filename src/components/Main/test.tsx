import { Main } from '.';

import { render, screen } from '@/utils/test';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /NextJS - boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#283036' });
  });
});
