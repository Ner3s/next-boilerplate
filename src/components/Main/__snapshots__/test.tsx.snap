// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-f1e8345d-0 fXhvcW"
>
  <h1
    class="sc-f1e8345d-1 dhdpus"
  >
    NextJS - Boilerplate
  </h1>
  <h2
    class="sc-f1e8345d-2 gagCGR"
  >
    NextJS, Typescript e Styled-Components
  </h2>
</main>
`;
